pandas/tests/generic/test_frame.py

```diff
@@ -47,19 +47,28 @@ def test_set_axis_name_mi(self, func):
 
     def test_nonzero_single_element(self):
         # allow single item via bool method
+        msg = (
+            "DataFrame.bool is now deprecated and will be removed "
+            "in future version of pandas and cases that relied on it will raise "
+            "a future warning"
+        )
         df = DataFrame([[True]])
-        assert df.bool()
+        df1 = DataFrame([[False]])
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            assert df.bool()
 
-        df = DataFrame([[False]])
-        assert not df.bool()
+        with tm.assert_produces_warning(FutureWarning, match=msg):
+            assert not df1.bool()
 
         df = DataFrame([[False, False]])
-        msg = "The truth value of a DataFrame is ambiguous"
-        with pytest.raises(ValueError, match=msg):
-            df.bool()
-        with pytest.raises(ValueError, match=msg):
+        msg1 = "The truth value of a DataFrame is ambiguous"
```

Comment:
Instead of `msg` and `msg1`, can you name these something more descriptive - I'd suggest `msg_warn` and `msg_err` respectivley.

Reply:
Done